src/DataPersister/MessengerDataPersister.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\DataPersister;
+
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Util\ClassInfoTrait;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+/**
+ * Dispatches the given resource using the message bus of Symfony Messenger.
+ *
+ * @experimental
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+final class MessengerDataPersister implements DataPersisterInterface
+{
+    use ClassInfoTrait;
+
+    private $resourceMetadataFactory;
+    private $messageBus;
+
+    public function __construct(ResourceMetadataFactoryInterface $resourceMetadataFactory, MessageBusInterface $messageBus)
+    {
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->messageBus = $messageBus;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports($data): bool
+    {
+        return true === $this->resourceMetadataFactory->create($this->getObjectClass($data))->getAttribute('messenger');
```

Comment:
Just double checking here: Are you sure you want to keep the `messenger` attribute a boolean? Or can you think of anything we might want to be able to configure at some point in the future? If so, I think we should turn that into an array for forward compat :)

Reply:
E.g. custom attributes for a general `AttributesStamp` on the envelope?  That way we could just pass arbitrary attributes with it that can be used in the message handlers.

Reply:
We'll be able to change that in the future without introducing a BC break.